pkg/installmanager/installmanager_test.go

```diff
@@ -863,3 +864,100 @@ spec:
 		})
 	}
 }
+
+func Test_patchAzureOverrideCreds(t *testing.T) {
+	var clusterInfraConfigBytes []byte = []byte(`---
+apiVersion: config.openshift.io/v1
+kind: Infrastructure
+status:
+  infrastructureName: hive-cluster-g7fqb
+  platformStatus:
+    azure:
+      cloudName: AzurePublicCloud
+      resourceGroupName: hive-cluster-g7fqb-rg
+`)
+
+	cases := []struct {
+		name                string
+		overrideSecretBytes []byte
+		expectModified      bool
+		expectErr           bool
+	}{
+		{
+			name: "Patch applies successfully",
+			overrideSecretBytes: []byte(`---
+apiVersion: v1
+data:
+  azure_client_id: YWFhCg==
+`),
+			expectModified: true,
+			expectErr:      false,
+		},
+		{
+			name: "Patch applies successfully with region exists and same",
+			overrideSecretBytes: []byte(`---
+apiVersion: v1
+data:
+  azure_region: Y2VudHJhbHVz
+`),
+			expectModified: true,
+			expectErr:      false,
+		},
+		{
+			name: "Patch fails due to region exists but different",
+			overrideSecretBytes: []byte(`---
+apiVersion: v1
+data:
+  azure_region: ZWFzdHVzCg==//eastus
```

Comment:
I don't think you're actually getting a comment here, as slashes are valid in both yaml and base64. By chance, since `//eastus` happens to have 8c, the whole value will decode without error to `eastus<binary garbage>`, which will still hit the desired code path since it doesn't match `centralus`. But it would better demonstrate your intent if you put the comment outside of the byte array.

Reply:
Thanks Eric, just fixed